core/trino-main/src/main/java/io/trino/split/SplitManager.java

```diff
@@ -52,7 +55,14 @@ public SplitManager(CatalogServiceProvider<ConnectorSplitManager> splitManagerPr
         this.splitManagerProvider = requireNonNull(splitManagerProvider, "splitManagerProvider is null");
         this.tracer = requireNonNull(tracer, "tracer is null");
         this.minScheduleSplitBatchSize = config.getMinScheduleSplitBatchSize();
-        this.executor = new BoundedExecutor(newCachedThreadPool(daemonThreadsNamed("splits-manager-callback-%s")), config.getMaxSplitManagerCallbackThreads());
+        this.executorService = newCachedThreadPool(daemonThreadsNamed("splits-manager-callback-%s"));
+        this.executor = new BoundedExecutor(executorService, config.getMaxSplitManagerCallbackThreads());
+    }
+
+    @PreDestroy
+    public void shutdown()
+    {
+        executorService.shutdown();
```

Comment:
What's the benefit of doing this? The underlying pool uses daemon threads, so they'll all terminate when the VM stops. `shutdown()` does not wait for or interrupt the threads, so it won't do much except for preventing new tasks from being submitted.

Reply:
In a typical deployment situation, this has no effect since the `SplitManager` is created only once. However, in tests this ensures that the threadpool is shutdown and the threads stopped earlier than they would if the usual cached threadpool 60 second idle timeout is reached.